### ISSUE TITLE
FreePIE+Opentrack issue

### ISSUE THREAD

**Issue by Gpinchon** [opentrack/opentrack#110]
Hi,
I'm having an issue when using FreePIE with my Smartphone. When I look up (pitch+) the view starts to rotate slowly, and when I'm looking straight forward it sometimes start to drift randomly...
The strange thing is that the raw data seems correct and is not changing at all, but the game data keeps slowly changing, Opentrack seems to slowly/randomly increment or decrement some values. Everytime I look up, when I look forward I have to re-center the view as it's rotated to yawn axis...
Also, if I shake my head, even if the received raw data indicates the view should be back at 0, the view is totally de-centered.

**Comment by Gpinchon** [comment]
I am using Accela. But after looking into the issue, I saw the drift only happened randomly...
I think it could be due to the phone recalibrating its Magnetic sensor when turning the head on all axis several times (like it's supposed to do)...
So I think there is no fix, except perhaps from the FreePIE creator to find a way to use an other sensor than the Magnetic Sensor for rotating...

**Comment by KyokushinPL** [comment]
@Gpinchon You mounting phone into headset with speakers?

**Comment by sthalik** [comment]
@Gpinchon for phones with gyro hardware it should work. There's tooling for Android apps that uses all three sensors and we're using that.

**Comment by Gpinchon** [comment]
@KyokushinPL Yep, I am mounting it with face up.
@sthalik mine has gyro and magnetic sensor, but I guess magnetic sensor keeps recalibrating when moving my head around various axis several times, I have not found any way to avoid that, I guess I am moving my head too much... :-/
Anyway, I stopped trying to get it to work properly, it's sad because it is very amazing to be able to move my head around without any input... But this recalibration thing making my head spin is quite obnoxious...

**Comment by KyokushinPL** [comment]
@Gpinchon maybe magnets in headset interfere magnetometer readings in smartphone.
Try it without smartphone.

My magnets interfere too, i can obey it with touching phone to magnets for about 20 seconds. I do not know why, but it helps, maybe phone is recalibrating magneto in some way.

**Comment by Gpinchon** [comment]
Well, last update fixed the transfered data, now they're correctly transmitted and no drift appears.
Just to be sure that no drift occures because of triggering the compass recalibration, I reduced the amount of head turning necessary by using a combo of headtracking and keys.
For instance, in Elite Dangerous, I use the UI zoom to access side panels instead of turning my head, so I don't rotate my head several times, triggering recalibration.
Also, I've found that when I strap my smartphone to my headset, I need to strap it with the top (the side with the power button) facing the computer, it makes any remaining drift disappear, it must have something to do with the compass location in the device...
